pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,408 @@
+          fi
+        done
+
+        IFS=" " read -ra last_snapshot <<< "${data[-1]}"
+
+        if [[ ${last_schedule_epoch_time} -gt ${last_snapshot[2]}  ]]; then
+          no_error=false
+          debug_msg+="[ERROR] opensearch-backup cronjob has not been scheduled\n  "
+        fi
+    else
+        no_error=false
+        debug_msg+="[ERROR] opensearch-backup cronjob doesn't exist\n"
+    fi
+  else
+    no_error=false
+    debug_msg=$(echo -e "$repo_exists_status" | jq)
+  fi
+
+
+  if $no_error; then
+    echo "success ✔"
+    echo "[DEBUG] All snapshots are either completed or in progress"
+  else
+    echo "failure ❌";
+    echo -e "$debug_msg"
+  fi
+}
+
+check_opensearch_indices(){
+  echo -ne "Checking opensearch indices status ... "
+  debug_msg=""
+  no_error=true
+
+  for index in 'other' 'kubernetes' 'kubeaudit' 'authlog'; do
+    res=$(curl -w "%{http_code}"  -o /dev/null -ksIL -u admin:"${adminPassword}" -X HEAD "https://opensearch.${opsDomain}/${index}")
+    if [[ $res != "200" ]]; then
+      debug_msg+="[ERROR] Missing index : ${index}\n"
+      no_error=false
+    fi
+  done
+
+  if $no_error; then
+    echo "success ✔"
+    echo -e "[DEBUG] All indices are present"
+  else
+      echo "failure ❌";
+      echo -e "$debug_msg"
+  fi
+}
+
+check_opensearch_breakers(){
+  echo -ne "Checking opensearch breakers ... "
+  breakers_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_nodes/_all/stats/breaker")
+  no_error=true
+  debug_msg=""
+  nodes_data=$(echo "$breakers_data" | jq ".nodes")
+  readarray -t nodes < <(jq -c 'to_entries[]' <<< "$nodes_data")
+
+  for node in "${nodes[@]}"; do
+    node_name=$(jq '.key' <<< "$node")
+    readarray -t breakers < <(jq -c '.value.breakers | to_entries[]' <<< "$node")
+    for breaker in "${breakers[@]}"; do
+      tripped=$(jq ".value.tripped" <<< "$breaker")
+      name=$(jq ".key" <<< "$breaker")
+      if [[ $tripped == "1" ]]; then
+        no_error=false
+        debug_msg+="[DEBUG] A circuit breaker : $name has been triggered for node: $node_name\n"
+      fi
+    done
+  done
+
+  if $no_error; then
+    echo "success ✔"
+    echo "[DEBUG] None of the circuit breakers has been triggered"
+  else
+    echo "failure ❌";
+    echo -e "$debug_msg"
+  fi
+
+}
+
+check_opensearch_aliases(){
+  echo -ne "Checking opensearch aliase indices mapping ... "
+  no_error=true
+  debug_msg=""
+
+  curl -sk -o /tmp/response -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/aliases"
+
+  aliases=$(awk '{print $1}' < /tmp/response)
+  aliases_arr=("$aliases")
+
+  uniq_aliases=$(echo "${aliases_arr[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' ')
+
+  for alias in $uniq_aliases; do
+    write_index_found=false
+    if ! [[ $alias =~ ^[\.*] ]]; then
+      alias_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_alias/${alias}")
```

Comment:
Suggestion to reduce the loops as they take quite some time with many indices.

```suggestion
      alias_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_alias/${alias}")
      
      # Note: Only set to true when found, empty otherwise
      is_write_index="$(echo "$alias_data" | jq ".[].aliases.$alias.is_write_index | select(. == true)")"
```